### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by NazarAli** [filamentphp/filament#9385] [General]
when edit record and back to table it not return to last page opened it return to first page 
how can solve this issue ?

**Comment by sinnbeck** [comment]
Use a modal? Or write your own redirect logic to go back to the correct page
https://filamentphp.com/docs/3.x/panels/resources/editing-records#customizing-redirects

**Reply by NazarAli** [reply]
i try this methods in Edit Page

``` 
   protected function getRedirectUrl(): string
    {
        return $this->previousUrl ?? $this->getResource()::getUrl('index');
    }
```

but not work it redirect to first page

**Reply by sinnbeck** [reply]
Weird. Just tested that code in my own app and it works just fine
```php
    protected function getRedirectUrl(): string
    {
        return $this->previousUrl ?? $this->getResource()::getUrl('view', ['record' => $this->getRecord()]);
    }
```
Maybe try updating to the latest filament with `composer update`

**Reply by NazarAli** [reply]
the problem how get page number and the page number not show in URL i think because it ajax request ?

**Reply by sinnbeck** [reply]
The url does not change when you click on a page link in the list? Mine appends something like `?page=10`

**Reply by NazarAli** [reply]
i disabled the spa in panel . and the page number not show in url . i used the latest update

![Untitled](https://github.com/filamentphp/filament/assets/7056487/084e5511-a837-43ae-9b6a-b44976c14db0)

**Reply by sinnbeck** [reply]
Yeah without it in the url, you cannot redirect back to it. What exactly did you change to have it removed from the url ?

**Reply by NazarAli** [reply]
the problem because i  add deferLoading to table that not show page number .
i have another question how can speed the table ?
i try disable debugbar and disable x-debuge and still the slow performance

**Reply by sinnbeck** [reply]
So its solved by turning that off again?

Use debugbar to see the queries being run, and then run them inside a database manager with "explain" in front of it. You can paste he query into this page to get an explanation of how your indexes are run.

Also make sure you have query caching on in mysql. Having it off can slow down mysql by alot

**Reply by NazarAli** [reply]
i not have in sql query 
but i see the count that used in pagination take more time . and if i have more than one million records it take more time
how can solve this ? 
if i enable the mysql cache can reduce the count time ? and how can enable mysql cache ?

![555](https://github.com/filamentphp/filament/assets/7056487/e2dbbd39-49e6-4c09-b27c-6dcd55ddbf2d)

**Reply by sinnbeck** [reply]
Yeah that is indeed a big count! Might be faster if you can split up your data into tabs. The cache might help some also

You need to check your mysql settings for `query_cache_type` which should be 1. You probably also want to increase `query_cache_size` to something like a few gigs of ram (depending on what specs your system has)

**Reply by sinnbeck** [reply]
You can also use simple pagination. That way it never does a count query
https://filamentphp.com/docs/3.x/tables/advanced#using-simple-pagination

But you dont get page selection. Just next/prev buttons

**Reply by NazarAli** [reply]
i think the problem in filament performance .
and when paginate take a lot of bandwidth. i use gzip to reduce the bandwidth

**Reply by sinnbeck** [reply]
A query is a query. Filament does not do anything to make that query slower. I am unsure what part of filament you think would make that count() query slow ?

**Reply by NazarAli** [reply]
now i used simple pagination and disable bulkActions because it want count 
total all query take 25ms and that good
but load page take more than 1s and i have 44 columns 

![Screenshot 2023-10-31 183800](https://github.com/filamentphp/filament/assets/7056487/16570476-f499-4065-bf72-362fa5e75b74)

**Reply by sinnbeck** [reply]
Maybe make a regular blade page with the same 20 rows of data and see how much faster it is. Mine loads instantly here
